department-of-veterans-affairs/caseflow
# Pamatya/appeals 31067

Resolves [APPEALS-31067](https://jira.devops.va.gov/browse/APPEALS-31067)

# Description
Please explain the changes you made here.
a) added a controller method that returns the serialized output for Event history.
b) added a  redux function to make request to the controller get the data and manage state. 
c) added an useEffect in ChangeHistory to fetch data. 



## Acceptance Criteria
- [ ] A controller method is returning right data. 
- [ ] redux is storing the correct form of data. 



## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. Login to the case flow VhaAdmin
3. Go to In Progress Tasks. 
4. Click on Claimant Name to go to disposition page. 
5. In Disposition page you should see 'View History'
6. you should be redirected to history page. 
7. Right click on the page and click on inspect.
8. You will see array listed in the console log.
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/3cd13380-4bbd-45dd-af9f-bf5d1295dd19)



# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
